src/nvc.rs

```diff
@@ -155,10 +257,19 @@ impl Nvc {
         self.psw_interrupt_mask_level = ((value as usize) >> 16) & 0x0f;
     }
 
+    pub fn fetch_instruction(&mut self, interconnect: &mut Interconnect, addr: u32) -> (u16, bool) {
```

Comment:
I think this should be called `fetch_instruction_halfword` to make it clear that it may not give you the full instruction (even though it's more verbose).